### PR TITLE
Fixes #16987 - Refactored subscription and content to facets

### PR DESCRIPTION
**Pull request by ShimShtein**
Now the facets are properly registered and can contribute
attributes to a host.

Should be tested with https://github.com/theforeman/foreman/pull/3987

### CODE REVIEW

**Comment by ShimShtein** [comment]
Consolidated facet registration code

File: lib/katello/plugin.rb [hunk 1]
```diff
@@ -203,6 +203,16 @@
                    :onlyif => proc { |proxy| proxy.has_feature?(SmartProxy::PULP_FEATURE) || proxy.has_feature?(SmartProxy::PULP_NODE_FEATURE) }
   end
 
+  register_facet Katello::Host::ContentFacet, :content_facet do
```
Comment (ShimShtein) [inline lib/katello/plugin.rb]: @lzap, one thing that I could not test was apipie generation. There was an issue with apipie that the documentation was generated before the plugin could register its facets, so Katello properties were omitted from Host documentation

**Comment by dLobatog** [comment]
The Foreman part of this is merged - so you can test with nighly now

**Comment by jlsherrill** [comment]
[test]

**Comment by ShimShtein** [comment]
Can we retest this PR? The tests are not available anymore...

**Comment by jlsherrill** [comment]
[test]

**Comment by ShimShtein** [comment]
Fixed the tests and a small issue in redefining nested attributes.

**Comment by dLobatog** [comment]
Please @Katello/katello-core take a look at this. Without this, 1.14-RC1 + Katello break assigning a hostgroup to a host, with `undefined method `inherited_attributes' for #<Class:0x0000000853add8>` as Foreman is unable to inherit the content/subscription properties of the facet.

**Comment by jlsherrill** [comment]
@dLobatog thanks for pointing that out, didn't realize there was an actual issue that this solved (rather than a pure refactor).  I've opened http://projects.theforeman.org/issues/17869 (copied from the foreman issue) to track this change and assigned it to 3.3

**Comment by jlsherrill** [comment]
ACK thanks @ShimShtein !

**Pull request merged**